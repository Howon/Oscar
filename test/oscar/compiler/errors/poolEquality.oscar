message testNow()


actor Tester() {

  receive = {
    | testNow() => {
        Println("Error");
      }
    | die() => {
      }
  }
}

def main() => unit = {
  pool<Tester> pool1 = spawn pool<Tester>({}, 3);
  pool<Tester> pool2 = spawn pool<Tester>({}, 3);
  bool b = (pool1 == pool2);
  message<testNow>() |>> pool1;
  message<die>() |>> pool1;
  message<die>() |>> pool2;
}
