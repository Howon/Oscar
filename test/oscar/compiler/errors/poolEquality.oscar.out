Error: Pools cannot be compared for equality
